src/config/inbuilt.rs

```diff
@@ -0,0 +1,80 @@
+// Network and CPU load have to be done with the *async magic*
+use crate::{
+    config::{system_stats::*, PollScriptVar, ScriptVar, VarSource::Function},
+    value::{PrimitiveValue, VarName},
+};
+use anyhow::*;
+use std::{collections::HashMap, time::Duration};
+
+pub fn get_inbuilt_vars() -> HashMap<VarName, ScriptVar> {
```

Comment:
This function has some formatting issues, but even if they are fixed it's only semiiii readable.
- i don't think you need to provide the return type in the lambda explicitly, but if you do, then you at least don't need to specify `Error` (i think), as it should know that it's talking about `anyhow::Result` which only needs one value.
- i'm not a fan of importing `VarSource::Function` globally. Either don't at all, and use `VarSource::Function` explicitly (which i'd likely be in favour of, as long as it doesn't add additional line wrapping), or use a local use
(you can use the `use` statement inside of `get_inbuilt_vars`)

Reply:
or in fact, not sure if this is a great idea, but what do you think about using this:
```rs
macro_rules! builtin_vars {
    ($interval:expr, $($name:literal => $fun:expr),*$(,)?) => {{
        hashmap! {
            $(
            VarName::from($name) => ScriptVar::Poll(PollScriptVar {
                name: VarName::from($name),
                command: VarSource::Function($fun),
                interval: $interval,
            })
            ),*
        }
    }}
}
```
which could result in sth like
```
let data = builtin_vars! { std::time::Duration::from_seconds(1000),
  "EWW_CORES" => || Ok(PrimitiveValue::from(format("{:.1}", cores::cores()))),
  "EWW_BATTERY" => || Ok(PrimitiveValue::from(
    match battery::get_battery_capacity() {
        Err(e) => {log::error!("Couldn't get the battery capacity: {:?}", e); f32::NAN }
        Ok(o) => o as f32,
    }
  ))
};
 ```
 not 100% sure if i like this, but it could definitely clean up the code a lot,.... what do you think?

Reply:
![image](https://user-images.githubusercontent.com/5300871/108748191-9ac5ab80-753e-11eb-9b37-ecc13e5db113.png)
expands very sanely, too 👍🏻 

(jk, but this is not an issue, it's just weirdly formatted)

Reply:
Yeah this looks great. 
I'll use your macro, thanks!